### ISSUE TITLE
Progress also displays x-factor (Download speed / playback speed = X) e.g. Downloading at 1.35MiB/s (= 3.4x)

### ISSUE THREAD

**Issue by porg** [yt-dlp/yt-dlp#11006]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a feature unrelated to a specific site
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

## Download Progress is currently shown like this

```
[download] Destination: My Movie.fdash-p0va0br3001270-1.mp4
[download]  58.8% of ~ 976.12MiB at 1.35MiB/s ETA 07:24 (frag 161/274)
```

## Proposal: Add X-factor in brackets of download speed

```
[download] Destination: My Movie.fdash-p0va0br3001270-1.mp4
[download]  58.8% of ~ 976.12MiB at 1.35MiB/s (= 3.4x) ETA 07:24 (frag 161/274)
```

- Download speed / Playback speed = X-Factor
  - Below 1 it means your bandwidth is not enough for streaming, because the download speed is slowr than the needed playback speed.
  - Above 1 tells you how much faster you can download than to watch in real time.
- From the streaming manifest and/or the media container header the overall bit rate of the media container is known.
  - E.g. 3200 kb/s divided by 8 = 0.4MiB/s
  - Download speed 1.35 / Playback speed 0.4 = 3.4x
  - Which means it downloads 3.4x times faster than it plays back
- Do we really need this?
  - The ETA is already a good metrics.
  - Nevertheless the x-factor gives you an estimate of a factor/proportion.
  - I propose it's off by default, so that default output does not get too noisy, but can be enabled with some kind of `--progress-*` argument, for those use cases which benefit from this metric.

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by porg** [comment]
Moved that long submission checklist into comment below. 

### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a feature unrelated to a specific site
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

See issue description. 

### Provide verbose output that clearly demonstrates the problem

Not applicable. 

### Complete Verbose Output

Not applicable.

**Comment by Grub4K** [comment]
Dont. Theres a reason its there, its explicitly stated to not remove it (that includes removing it from the op) and theres a button to hide it. Please move it back

**Comment by porg** [comment]
1) I have moved it back.
2) I don't see any button to "hide all the overhead" and see the proposal text only.

**Comment by Grub4K** [comment]
2. You are right, I confused it with the PR template; Ive brought up the idea to expose it for issues as well internally.

An easy way could be to expose it as a key in the progress dict. That way users can add it easily through placing ` (=%(rate)2.1f)` or equivalent in `--progress-template`, and we dont have to add an extra switch.

**Comment by porg** [comment]
- Integrating a placeholder into the `--progress-template` would be a very clean approach. I also had that thought. You coming up with it too, confirms that this is the mental model of the user where to potentially expect it.
- When implementing it please also mention that new placeholder in the manpage / readme.
- Was my technical assumption correct that yt-dlp can indeed easily determine the overall bit rate of the media container ?
> From the streaming manifest and/or the media container header the overall bit rate of the media container is known.
- Will you implement it?

**Comment by porg** [comment]
## Regarding the feature request template

1. IMHO the template is currently overly formal and over-structured for a feature request template, which by its very nature tries to describe something new, hence no need to refer to so much env/debug output etc.
    - Where appropriate/relevant the responsible submitter will provide the necessary background/environment info.

2. If you insist on keeping it that way, maybe you can leave the field order as-is (to force the submitter to consciously go through them) but make the output so that the proposal itself:
    a. Comes first
    b. Allows to use H1 and lower, due to not being sandwiched in a H3 hence only allowing H4 and deeper
    c. Followed by `## FEATURE REQUEST SUBMISSION METADATA - DO NOT DELETE` which then contain all the rest.
    - Or you manage to offer a "show/hide details" button (does Github allow these? or provide functions for that?)

**Comment by grqz** [comment]
> 2\. * Or you manage to offer a "show/hide details" button (does Github allow these? or provide functions for that?)

see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections
for example,
<details><summary>The text inside the summary node</summary>

[inside details, outside summary]
more details
(can use markdown as usual)
```
(inside codeblock)
```

</details>
generated from:

~~~MD
<details><summary>The text inside the summary node</summary>

[inside details, outside summary]
more details
(can use markdown as usual)
```
(inside codeblock)
```

</details>
~~~

**Comment by bashonly** [comment]
> Was my technical assumption correct that yt-dlp can indeed easily determine the overall bit rate of the media container ?

No, this is not necessarily true. It could *estimate* the bitrate based on the duration provided by the website metadata and the total filesize or fragment size, but it does not probe the media file as it downloads. And e.g. if it's a single non-`faststart` mp4 file, it would be almost impossible to do so
